bin/node/src/commands.rs

```diff
@@ -0,0 +1,168 @@
+use crate::chain_spec::{self, get_account_id_from_seed, AuthorityKeys, LOCAL_AUTHORITIES};
+use crate::cli::{Cli, ExtraParams};
+use aleph_primitives::AuthorityId as AlephId;
+use log::info;
+use sc_cli::{CliConfiguration, Error, KeystoreParams, SharedParams};
+use sc_keystore::LocalKeystore;
+use sc_service::config::{BasePath, KeystoreConfig};
+use sp_application_crypto::key_types;
+use sp_consensus_aura::sr25519::AuthorityId as AuraId;
+use sp_core::crypto::KeyTypeId;
+use sp_core::{ed25519, sr25519};
+use sp_keystore::SyncCryptoStore;
+use std::io::Write;
+use std::sync::Arc;
+use structopt::StructOpt;
+
+/// The `bootstrap-chain` command is used generate keys for the genesis authorities
+/// keys are written to the keystore of the authorities
+/// and the chain specification is printed to stdout in the JSON format
+#[derive(Debug, StructOpt)]
+pub struct BootstrapChainCmd {
+    /// Force raw genesis storage output.
+    #[structopt(long = "raw")]
+    pub raw: bool,
+
+    #[structopt(flatten)]
+    pub keystore_params: KeystoreParams,
+
+    #[structopt(flatten)]
+    pub shared_params: SharedParams,
+}
+
+impl CliConfiguration for BootstrapChainCmd {
+    fn shared_params(&self) -> &SharedParams {
+        &self.shared_params
+    }
+}
+
+impl BootstrapChainCmd {
+    fn upsert_aura_key(
+        &self,
+        keystore: &LocalKeystore,
+        key_type: KeyTypeId,
+    ) -> Result<AuraId, Error> {
+        let key = SyncCryptoStore::sr25519_public_keys(&*keystore, key_type)
+            .into_iter()
+            .next()
+            .map_or_else(
+                || {
+                    SyncCryptoStore::sr25519_generate_new(&*keystore, key_type, None)
+                        .map_err(|_| Error::KeyStoreOperation)
```

Comment:
We wrap the value in `Err` only to unwrap it below. Why not call `expect` on the result of calling `sr25519_generate_new`? Then `map_or_else` would be replaced with `unwrap_or_else`.

Reply:
good spot, refactored